### ISSUE TITLE
Hard-to-reproduce CTD after fast travel/cell transition

### ISSUE THREAD

**Issue by SteveTownsend** [SteveTownsend/SmartHarvestSE#199]
Nexus report:

> 01 September 2020, 1:18PM
> As of 2.9.3.0 I'm experiencing a CTD after fast traveling. Any ideas? The crash does not occur when I disable Smart Harvest SE. My crash logs show the following:
> 
> Possible relevant objects (4)
> {
> [1] TESNPC(Name: `Peth`, FormId: 00000007, File: `Skyrim.esm`)
> [1] PlayerCharacter(FormId: 00000014, BaseForm: TESNPC(Name: `Peth`, FormId: 00000007, File: `Skyrim.esm`))
> [ 19] BGSLocation(Name: `Katla's Farm`, FormId: 00018E38, File: `Unofficial Skyrim Special Edition Patch.esp <- HearthFires.esm <- Skyrim.esm`)
> [ 50] TESObjectCELL(Name: CRFFrostRiverFarmWest, FormId: 00009367, File: `Cutting Room Floor.esp`)
> }
> 
> Probable callstack
> {
> [0]0x7FF901BC83AC (SmartHarvestSE.dll+383AC)
> [1]0x7FF901BF9C44 (SmartHarvestSE.dll+69C44)
> [2]0x7FF901B9799C (SmartHarvestSE.dll+799C)
> [3]0x7FF9489D36EE (MSVCP140.dll+136EE)
> [4]0x7FF901C24A10 (SmartHarvestSE.dll+94A10)
> [5]0x7FF9489D2C8D (MSVCP140.dll+12C8D)
> [6]0x7FF9489D356F (MSVCP140.dll+1356F)
> [7]0x7FF901BF946A (SmartHarvestSE.dll+6946A)
> [8]0x7FF901C23A84 (SmartHarvestSE.dll+93A84)
> [9]0x7FF9489D372E (MSVCP140.dll+1372E)
> [10] 0x7FF901BF9801 (SmartHarvestSE.dll+69801)
> [11] 0x7FF901BF9049 (SmartHarvestSE.dll+69049)
> [12] 0x7FF901BFA37E (SmartHarvestSE.dll+6A37E)
> [13] 0x7FF959C80E82 (ucrtbase.dll+20E82)
> [14] 0x7FF95A6C7BD4 (KERNEL32.DLL+17BD4)
> [15] 0x7FF95C52CE51 (ntdll.dll+6CE51)
> }
> 
> Registers
> {
> AX: 0x2100000000 (u64):[141733920768]
> BX: 0x0 (NULL)
> CX: 0x2BF072B2500(void*)
> DX: 0x2BF072B2500(void*)
> SI: 0xD5680FF5E0 (char*) "?%"
> DI: 0xD5680FF490 (void*)
> BP: 0xD5680FF4A9 (void*)
> SP: 0xD5680FF3C0 (void*)
> IP: 0x7FF901BC83AC (SmartHarvestSE.dll+383AC) (void*)
> R8: 0x0 (NULL)
> R9: 0x7FF9489A0000 (VCRUNTIME140.dll+0) (char*) "MZ?"
> R10:0x7FF901D28328 (SmartHarvestSE.dll+198328) (void*)
> R11:0xD5680FF470 (void*)
> R12:0x2BE87240600(PlayerCharacter*) -> (FormId: 00000014, BaseForm: TESNPC(Name: `Peth`, FormId: 00000007, File: `Skyrim.esm`))
> R13:0x2BF072B2500(void*)
> R14:0x2BF072B2500(void*)
> R15:0x0 (NULL)
> Flags: 0x10202
> XMM0: (double)5.19684332961924E-315 / (float)0.3476563
> XMM1: (double)5.6365513197515E-315 / (float)512
> XMM2: (double)0 / (float)0
> XMM3: (double)0 / (float)0
> XMM4: (double)0 / (float)0
> XMM5: (double)0 / (float)0
> XMM6: (double)2 / (float)0
> XMM7: (double)5 / (float)0
> XMM8: (double)0.1 / (float)-1.588187E-23
> XMM9: (double)0 / (float)0
> XMM10: (double)0 / (float)0
> XMM11: (double)0 / (float)0
> XMM12: (double)0 / (float)0
> XMM13: (double)0 / (float)0
> XMM14: (double)0 / (float)0
> XMM15: (double)0 / (float)0
> }
> 
> Stack
> {
> [SP+0] 0x2C01D138AA0(void*)
> [SP+8] 0x7FF901BF9C44 (SmartHarvestSE.dll+69C44) (void*)
> [SP+10]0x2C0F7E67870(void*)
> [SP+18]0x2BFE00F6BF0(void*)
> [SP+20]0xD5680FF490 (void*)
> [SP+28]0x0 (NULL)
> [SP+30]0x3D16D2769B8(u64):[4198014347704]
> [SP+38]0x7FF901B9799C (SmartHarvestSE.dll+799C) (void*)
> [SP+40]0x112E0BE826D694B3 (u64):[1237940039285380275]
> [SP+48]0x311CDAD160000 (u64):[864000000000000]
> [SP+50]0x1 (u8):[1]
> [SP+58]0x7FF9489D36EE (MSVCP140.dll+136EE) (void*)
> [SP+60]0x0 (NULL)
> [SP+68]0xBF0605FA8462 (u64):[210032591012962]
> [SP+70]0x2C0010B39B8(void*)
> [SP+78]0x7FF901C24A10 (SmartHarvestSE.dll+94A10) (void*)
> [SP+80]0x0 (NULL)
> [SP+88]0x2BF05883C20(BGSLocation*) -> (Name: `Katla's Farm`, FormId: 00018E38, File: `Unofficial Skyrim Special Edition Patch.esp <- HearthFires.esm <- Skyrim.esm`)
> [SP+90]0x0 (NULL)
> [SP+98]0xD5680FF5E0 (char*) "?%"
> [SP+A0]0x17DCE00000000 (u64):[419798693445632]
> [SP+A8]0xD5680FF5E0 (char*) "?%"
> [SP+B0]0x17DCEA35A8A00 (u64):[419801434065408]
> [SP+B8]0x7FF9489D2C8D (MSVCP140.dll+12C8D) (void*)
> [SP+C0]0x0 (NULL)
> [SP+C8]0xF (u8):[15]
> [SP+D0]0x0 (NULL)
> [SP+D8]0x7FF9489D356F (MSVCP140.dll+1356F) (void*)
> [SP+E0]0x989680 (u32):[10000000]
> [SP+E8]0x2BE14D52460(u64):[3015416554592]
> [SP+F0]0x5E7F492DFFC2 (u64):[103900781608898]
> [SP+F8]0xD5680FF5C0 (char**) "SHSE_Logger"
> [SP+100] 0x17DCE67BFC124 (u64):[419800434065700]
> [SP+108] 0x7FF901BF946A (SmartHarvestSE.dll+6946A) (void*)
> [SP+110] 0xBF0605FA83E2 (u64):[210032591012834]
> [SP+118] 0x2C0010B3930(void*)
> [SP+120] 0x0 (NULL)
> [SP+128] 0x2C0010B3930(void*)
> [SP+130] 0x2C0010B39B8(void*)
> [SP+138] 0x0 (NULL)
> [SP+140] 0xD5680FF5D9 (void*)
> [SP+148] 0x7FF901C23A84 (SmartHarvestSE.dll+93A84) (void*)
> [SP+150] 0x2BF05883C20(BGSLocation*) -> (Name: `Katla's Farm`, FormId: 00018E38, File: `Unofficial Skyrim Special Edition Patch.esp <- HearthFires.esm <- Skyrim.esm`)
> [SP+158] 0x2BE23548C30(void*)
> [SP+160] 0x2BE23548C30(void*)
> [SP+168] 0x0 (NULL)
> [SP+170] 0xD5680FF501 (void*)
> [SP+178] 0x0 (NULL)
> [SP+180] 0x2BF0789FA40(TESObjectCELL*) -> (Name: CRFFrostRiverFarmWest, FormId: 00009367, File: `Cutting Room Floor.esp`)
> [SP+188] 0x2BE23548C30(void*)
> [SP+190] 0x2BF072B2500(void*)
> [SP+198] 0x7FF9489D372E (MSVCP140.dll+1372E) (void*)
> [SP+1A0] 0x2BE20DEF5C8(char*) "SHSE_Logger"
> [SP+1A8] 0xB (u8):[11]
> [SP+1B0] 0x7FF901D2D888 (SmartHarvestSE.dll+19D888) (char*) "UI/controls now good-to-go, wait before first scan"
> [SP+1B8] 0x32(u8):[50]
> [SP+1C0] 0x0 (NULL)
> [SP+1C8] 0x0 (NULL)
> [SP+1D0] 0x0 (NULL)
> [SP+1D8] 0x2C0010B39B8(void*)
> [SP+1E0] 0x0 (NULL)
> [SP+1E8] 0x0 (NULL)
> [SP+1F0] 0x0 (NULL)
> [SP+1F8] 0x3A10E32BA4 (u64):[249391426468]
> [SP+200] 0x2BE20DEF5C8(char*) "SHSE_Logger"
> [SP+208] 0xB (u8):[11]
> [SP+210] 0x17DCD00000002 (u64):[419794398478338]
> [SP+218] 0x38CEA45E0A4A4A(u64):[15989804043487818]
> [SP+220] 0x25A0 (u16):[9632]
> [SP+228] 0x0 (NULL)
> [SP+230] 0x0 (NULL)
> [SP+238] 0x0 (NULL)
> [SP+240] 0xBF0605FA82B2 (u64):[210032591012530]
> [SP+248] 0x0 (NULL)
> [SP+250] 0x7FF901D2C670 (SmartHarvestSE.dll+19C670) (char*) "SmartHarvestSE.esp"
> [SP+258] 0x0 (NULL)
> [SP+260] 0x0 (NULL)
> [SP+268] 0x7FF901D2D8C0 (SmartHarvestSE.dll+19D8C0) (char*) "????????BaseObjectBlocked"
> [SP+270] 0x2 (u8):[2]
> [SP+278] 0x7FF901BF9801 (SmartHarvestSE.dll+69801) (void*)
> [SP+280] 0x2 (u8):[2]
> [SP+288] 0x1 (u8):[1]
> [SP+290] 0x0 (NULL)
> [SP+298] 0x2C04E6E25C0(BGSRefAlias**)
> [SP+2A0] 0x0 (NULL)
> [SP+2A8] 0x0 (NULL)
> [SP+2B0] 0x0 (NULL)
> [SP+2B8] 0x0 (NULL)
> [SP+2C0] 0x0 (NULL)
> [SP+2C8] 0x0 (NULL)
> [SP+2D0] 0x7FF901D2C670 (SmartHarvestSE.dll+19C670) (char*) "SmartHarvestSE.esp"
> [SP+2D8] 0x12(u8):[18]
> [SP+2E0] 0x6C61767265746E00 (u64):[7809653462537629184]
> [SP+2E8] 0x73646E6F636553(u64):[32480047799690579]
> [SP+2F0] 0x0 (NULL)
> [SP+2F8] 0xF (u8):[15]
> [SP+300] 0x0 (NULL)
> [SP+308] 0x0 (NULL)
> [SP+310] 0x0 (NULL)
> [SP+318] 0x0 (NULL)
> [SP+320] 0x0 (NULL)
> [SP+328] 0x0 (NULL)
> [SP+330] 0x0 (NULL)
> [SP+338] 0x0 (NULL)
> [SP+340] 0x0 (NULL)
> [SP+348] 0x0 (NULL)
> [SP+350] 0x0 (NULL)
> [SP+358] 0x0 (NULL)
> [SP+360] 0x2BFE8E80160(void*)
> [SP+368] 0x7FF901BF9049 (SmartHarvestSE.dll+69049) (void*)
> [SP+370] 0x4000000000000000 (u64):[4611686018427387904] (f64):[2]
> [SP+378] 0x0 (NULL)
> [SP+380] 0x7548C2FFFF0F (u64):[128955369651983] (f32):[-127.9982]
> [SP+388] 0x2C04E6E25C0(BGSRefAlias**)
> [SP+390] 0x2BFE6B5B600(void*)
> [SP+398] 0x7FF901BFA37E (SmartHarvestSE.dll+6A37E) (void*)
> [SP+3A0] 0x0 (NULL)
> [SP+3A8] 0x0 (NULL)
> [SP+3B0] 0x0 (NULL)
> [SP+3B8] 0x0 (NULL)
> [SP+3C0] 0x2C106A0F630(void*)
> [SP+3C8] 0x7FF959C80E82 (ucrtbase.dll+20E82) (void*)
> [SP+3D0] 0x0 (NULL)
> [SP+3D8] 0x2C106A0F630(void*)
> [SP+3E0] 0x0 (NULL)
> [SP+3E8] 0x0 (NULL)
> [SP+3F0] 0x0 (NULL)
> [SP+3F8] 0x7FF95A6C7BD4 (KERNEL32.DLL+17BD4) (void*)
> [SP+400] 0x0 (NULL)
> [SP+408] 0x0 (NULL)
> [SP+410] 0x0 (NULL)
> [SP+418] 0x0 (NULL)
> [SP+420] 0x0 (NULL)
> [SP+428] 0x7FF95C52CE51 (ntdll.dll+6CE51) (void*)
> [SP+430] 0x0 (NULL)
> [SP+438] 0x0 (NULL)
> [SP+440] 0x0 (NULL)
> [SP+448] 0x0 (NULL)
> [SP+450] 0x0 (NULL)
> [SP+458] 0x0 (NULL)
> [SP+460] 0x4174120A00000000 (u64):[4716414543922724864]
> [SP+468] 0x0 (NULL)
> [SP+470] 0x0 (NULL)
> [SP+478] 0x7FF95953F400 (KERNELBASE.dll+FF400) (void*)
> [SP+480] 0xD5680FE4C0 (void*)
> [SP+488] 0x8280001FFE4609F9 (i64):[-9043227914349966855]
> [SP+490] 0x7FF91827E60A (u64):[140707828852234]
> [SP+498] 0xD5680FE4C0 (void*)
> [SP+4A0] 0x0 (NULL)
> [SP+4A8] 0x0 (NULL)
> [SP+4B0] 0x0 (NULL)
> 
> (The stack list goes on for quite a while with 0x0 (NULL) entries)
> 
> 01 September 2020, 1:43PM
> Ok well it seems to have resolved now. I went and ran around for a while, entered a few new areas manually, and when I fast traveled again there was no crash. Maybe SHSE just had to figure some stuff out?

**Comment by SteveTownsend** [comment]
2nd Nexus report:

> Smart Harvest Log
> https://pastebin.com/UkwTqCP3
> .net Crash Log
> https://pastebin.com/GywS5QW1
> Going from Area in the Rift (cell 0x000097d2 to Whiterun)

**Comment by SteveTownsend** [comment]
The stack trace from LogStack Walker is more accurate than .Net Script Framework:
```
ERROR: SymGetLineFromAddr64, GetLastError: 487 (Address: 00007FF8ACC983AC)
ERROR: SymGetLineFromAddr64, GetLastError: 487 (Address: 00007FF8ACCF4A10)
ERROR: SymGetLineFromAddr64, GetLastError: 487 (Address: 00007FF8ACCF3A84)
ERROR: SymGetLineFromAddr64, GetLastError: 487 (Address: 00007FF8ACCC9801)
ERROR: SymGetLineFromAddr64, GetLastError: 487 (Address: 00007FF8ACCC9049)
ERROR: SymGetLineFromAddr64, GetLastError: 487 (Address: 00007FF8ACCCA37E)
ERROR: SymGetLineFromAddr64, GetLastError: 487 (Address: 00007FF8FF5D1542)
ERROR: SymGetLineFromAddr64, GetLastError: 487 (Address: 00007FF901BF6FD4)
ERROR: SymGetLineFromAddr64, GetLastError: 487 (Address: 00007FF901E5CEC1)
00007FF901E5CEC1 (ntdll): (filename not available): RtlUserThreadStart
```

**Comment by SteveTownsend** [comment]
Fix in v2.9.4.0 seems to have worked
